canaanwest

```diff
@@ -0,0 +1,198 @@
+#PLANETS
+class Planets
+  attr_reader :name, :rating, :population, :distance_from_sun, :year_length, :qualities
+  attr_accessor :radius, :avg_temp, :show_qualities
+
+  def initialize(name, population, rating, distance_from_sun, year_length)
+    @name = name
+    @population = population
+    @rating = rating
+    @distance_from_sun = distance_from_sun
+    @year_length = year_length
+    @qualities = { name: @name, population: @population, rating: @rating, distance_from_sun: @distance_from_sun, year_length: @year_length }
+  end
+
+  def show_qualities
+    my_qualities = ""
+    @qualities.each do |quality, value|
+      my_qualities += "#{quality.capitalize}:  #{value}\n"
+    end
+    return "#{my_qualities} \n"
+  end
+
+  def add_quality(key, value)
+    @qualities[key.to_sym] = value
+  end
+
+  def distance_from=planet_compare_name
+    return find_planet(planet_compare_name).distance_from_sun - self.distance_from_sun
+  end
+end #end planets
+
+#SOLAR SYSTEM
+class SolarSystem
+  attr_accessor :planets
+  attr_reader :age
+
+  def initialize(array_of_planets, age)
+    @planets = array_of_planets
+    @age = age
+  end
+
+  def add_planet=planet
+    @planets.push(planet)
+  end
+
+  def planet_attributes
+    @planets.each do |planet|
+      puts "#{planet.show_qualities}"
+    end
+  end
+
+  def print_planet_names
+    i = 0;
+    planet_names = ""
+    @planets.each do |planet|
+      planet_names += "#{i + 1}. #{planet.name} \n"
+      i += 1
+    end
+    return planet_names
+  end
+
+  def find_planet(input_name)
+    @planets.each do |planet|
+      if planet.name == input_name
+        return planet
+      end
+    end
+  end
+end #end of Class SolarSystem
+
+
+def user_make_planet
+  puts "\nWhat's the planet's name?"
+  name = gets.chomp.downcase
+
+  puts "\nWhat's the population?"
+  population = gets.chomp.to_i
+
+  puts "\nWhat's the rating?"
+  rating = gets.chomp.to_f
+
+  puts "\nWhat's the distance from the sun?"
+  distance = gets.chomp.to_f
+
+  puts "\nWhat's the length of a year?"
+  year_length = gets.chomp.to_f
+
+  user_planet = Planets.new(name, population, rating, distance, year_length)
+
+  puts "\nYour planet was created! See it in the list?\n"
+  return user_planet
+end
+
+def user_compare_planets(solarsystem, planet1, planet2)
+  planet1_position = solarsystem.find_planet(planet1)
```

Comment:
very small observation: your code doesn't check if the input name is a valid choice before you call `solarsystem.find_planet` :P